ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -223,13 +230,17 @@ public PullQueryResult execute(
           contextStacker,
           pullQueryMetrics);
 
-      return handlePullQuery(
+      final PullQueryResult result = handlePullQuery(
           statement,
-          executionContext,
+          ksqlEngine,
           serviceContext,
           pullQueryContext,
           routingOptions
       );
+
+      pullQueryMetrics.ifPresent(metrics ->
+          metrics.recordLatency(Time.SYSTEM.nanoseconds() - startTimeNanos));
```

Comment:
Why not just have:
```suggestion
      pullQueryMetrics.ifPresent(metrics ->
          metrics.recordLatency(startTimeNanos));
```

And change `recordLatency` to take the end time in ns, encapsulating the `Time.SYSTEM.nanoseconds()` into a single place...?